cou/utils/juju_utils.py

```diff
@@ -499,3 +509,27 @@ async def _wait_for_active_idle() -> None:
             apps = await self._get_supported_apps()
 
         await _wait_for_active_idle()
+
+    async def get_model_machines(self) -> dict[str, Machine]:
+        """Get all the machines in the model.
+
+        :return: Dictionary of the machines found in the model. E.g: {'0': Machine0}
+        :rtype: dict[str, Machine]
+        """
+        juju_applications = await self.get_applications()
+        machines: dict[str, Machine] = {}
+        for app in juju_applications.values():
+            charm_name = app.charm_name
+            for unit in app.units:
+                unit_machine = unit.machine.data
```

Comment:
I think it would be easier if we can do (not sure if it's possible):
```python
for juju_machine in model.machines:
    is_data_plane = any(app.charm_name in DATA_PLANE_CHARMS for app in juju_machine.apps)
    machine = Machine(
        machine_id=juju_machine["id"],
        hostname=juju_machine["hostname"],
        az=juju_machine["hardware-characteristic"].get("availability-zone"),
        is_data_plane=is_data_plane
    )
    ...
```